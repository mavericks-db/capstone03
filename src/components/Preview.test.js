import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Preview from './Preview';
import store from './redux/configureStore';

const MockPreview = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Preview pokemon={{ id: '1', name: 'bulbasaur' }} />
    </BrowserRouter>
  </Provider>
);

describe('Test the Preview component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<MockPreview />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders link', () => {
    render(<MockPreview />);
    const linkElement = screen.getByRole('link', {
      name: /01 bulbasaur/i,
    });
    expect(linkElement).toBeInTheDocument();
  });

  test('renders link', () => {
    render(<MockPreview />);
    const headingElement = screen.getByRole('heading', {
      name: /bulbasaur/i,
    });
    expect(headingElement).toBeInTheDocument();
  });
});
