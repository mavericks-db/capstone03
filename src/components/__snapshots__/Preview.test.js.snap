// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Preview component renders correctly 1`] = `
<a
  className="box"
  href="/pokemon/details/1"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "hsl(219,45%,49%)",
    }
  }
>
  <div
    className="oval"
  >
    <h1>
      01
    </h1>
    <img
      alt=""
      src="pokeball.png"
    />
    <h2>
      BULBASAUR
    </h2>
  </div>
</a>
`;
